### DISCUSSION TITLE
How to automatically set the 'inserted Id' of the parent to the collection of referencing childs ?

### DISCUSSION THREAD

**Discussion by youssefbennour** [ServiceStack/Discuss#83] [Q&A] [answered]
Hello,
I have a scenario similar to the below :
```
public class Parent {
        [PrimaryKey, AutoIncrement]
        public int Id { get; set; }
        [Reference]
        public HashSet<Child> Children { get; set; }
}
public class Child {
        [AutoIncrement, PrimaryKey]
        public int Id { get; set; }
        public int ParentId { get; set;}
        [Reference]
        public required Parent Parent { get; set; }
}
```
    
  I want to insert an instance of the 'Parent' with it's 'Child' elements at once using Save(parentInstance, references: true)
  but I get an error saying "Cant find 'ParentId' Property on Type Hashset.....
  I thought that on Save(parentInstance, references: true) the 'ParentId' property in 'Child' elements will be set automatically, but as it appears to me that's not what's happening.
  Am I missing somehting ?

**Comment by mythz** [comment, marked as answer]
Please read the docs for the feature you're using, if what you're trying to use doesn't exist in the docs (or anywhere else, e.g. Forums / StackOverflow / Source Code, etc) it's not supported. 

[Poco References](https://docs.servicestack.net/ormlite/reference-support) need to use either `Type` for **1:1** relationships or `List<Type>` for **1:M** relationships.
